Ds3/Helpers/IBaseJob.cs

```diff
@@ -94,5 +94,10 @@ public interface IBaseJob<TSelf, TItem> where TSelf : IBaseJob<TSelf, TItem>
         /// Fires handlers with an object name and the meta-data associated with it.
         /// </summary>
         event Action<string, IDictionary<string, string>> MetadataListener;
+
+        /// <summary>
+        /// Fires handlers with an object name, offset and the exception thrown
```

Comment:
Sure, but it should at the very least state that a failure is something that we cannot recover from in the helper functions.